playpen/coverage/coverage_hook

```diff
@@ -0,0 +1,488 @@
+"""
+import argparse
+import hashlib
+import glob
+import os
+import shutil
+import sys
+
+import coverage.files
+from distutils.sysconfig import get_python_lib
+
+# a list consisting of either an importable module name or a path to module that contains
+# a pulp entry point that doesn't get covered with the generic pulp package hook.
+# this includes modules containing WSGI applications, celery workers, or other services.
+# if coverage is suspiciously missing for a service, this list probably needs a new item
+_entry_modules = [
+    # celerty task workers
+    'pulp.server.async.app',
+    # wsgi app modules -- the wsgi files themselves don't always work as hook points;
+    # the sitecustomize import needs to happen in the file where the wsgi application is def'd.
+    'pulp.server.content.web',
+    'pulp_puppet.forge.wsgi',
+    'pulp.repoauth.wsgi',
+    'pulp.server.webservices.application',
+    # streamer auth application and server
+    '/usr/share/pulp/wsgi/streamer_auth.wsgi',
+    '/usr/share/pulp/wsgi/streamer.tac',
+]
+
+# help strings for the argument parser subparsers
+_install_help = """\
+install the coverage-enabling sitecustomize file to a site-wide python packages dir
+
+If the sitecustomize file already exists and is not the one installed
+by this script, an error will be raised unless --force is used.
+
+If --deps is specified, install the requirements listed in coverage-requirements.txt
+"""
+_uninstall_help = """\
+remove the coverage-enabling sitecustomize file from a site-wide python packages dir
+
+If the sitecustomize file is not the one installed by this script, an
+error will be raised unless --force is used.
+
+If --deps is specified, uninstall the requirements listed in coverage-requirements.txt
+"""
+_report_help = """\
+report the saved coverage reports and write the output to the specified directory
+
+report will fail if coverage has not yet been installed
+"""
+_root_epilog = "This script is intended to be run with root privileges."
+
+
+def _files():
+    """
+    files used by both install and uninstall
+
+    DRY function since we need these filenames for multiple subcommands
+    """
+    this_script = os.path.abspath(__file__)
+    script_dir = os.path.dirname(this_script)
+    requirements = os.path.join(script_dir, 'coverage-requirements.txt')
+    pulp_sitecustomize = os.path.join(script_dir, '_sitecustomize.py')
+    target_sitecustomize = os.path.join(get_python_lib(), 'sitecustomize.py')
+    return {
+        'pulp_coverage': this_script,
+        'pulp_coverage_requirements': requirements,
+        'pulp_sitecustomize': pulp_sitecustomize,
+        'target_sitecustomize': target_sitecustomize,
+    }
+
+
+def _hash(filename):
+    """
+    given a filename, return the string hash of a file
+
+    Used in this module to determine if an installed sitecustomize file
+    is the same one to install/uninstall.
+
+    Args:
```

Comment:
I think I like this way, you might have me convinced. However, we have been doing it the other way consistently for a while now. Would you be willing to do it the way specified in our contributor docs until you convince the team to change it?